typescript/packages/orb/src/index.test.ts

```diff
@@ -0,0 +1,134 @@
+import { NoosphereContext } from './index.js';
+import { expect } from '@esm-bundle/chai';
+import { SphereFile } from './noosphere.js';
+
+function makeRandomName(prefix: string, randomCharacters: number = 16) {
+  const alphabet =
+    'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+  const alphabetLength = alphabet.length;
+
+  for (let i = 0; i < randomCharacters; ++i) {
+    prefix += alphabet.charAt(Math.floor(alphabetLength * Math.random()));
+  }
+
+  return prefix;
+}
+
+describe('orb', () => {
+  it('can initialize a NoosphereContext', () => {
+    const noosphere = new NoosphereContext('foo');
+
+    noosphere.free();
+  });
+
+  it('can create a key', async () => {
+    const noosphere = new NoosphereContext('foo');
+    const keyName = makeRandomName('key');
```

Comment:
I think it'd be valuable confirming that non-ascii/UTF8 JS strings work as intended over wasm (as opposed to a TextDecoder + byte writing method)

Reply:
SG will verify

Reply:
Good thing you asked about this, there are some considerations to be made. See: https://rustwasm.github.io/docs/wasm-bindgen/reference/types/str.html#utf-16-vs-utf-8

TL;DR the conversion is mostly lossless, but lossy around the edges.

I think in the end we probably would want to stick with the two methods as written, since they give our users two options with different trade-offs. If a user wants to control the conversion, they can encode the bytes themselves on the JS side and pass them in. If they are okay with the conversion as performed by our codegen, then they can pass a string.

I'll add comments to this change to advise of the caveats.

Reply:
SGTM, I think documenting the string method with the rust wasm encoding note is sufficient